### PR TITLE
Monitors store copy on server and client individually

### PR DESCRIPTION
**Pull request by tinwelint**
so that
BackupServiceIT#shouldContainTransactionsThatHappenDuringBackupProcess can
be implemented correctly and test what it was initially designed to test,
namely that a store copy gets correct even in the face of changing store
files while copying.

Also fixed a transaction log header issue where the txId would be 0
instead of 1 (base txId) if no transaction streamed.

thanks to @davidegrohmann

### CODE REVIEW

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/SwitchToSlaveTest.java [hunk 1]
```diff
@@ -132,7 +134,7 @@ private static SwitchToSlave newSwitchToSlaveSpy()
                 mock( ClusterMemberAvailability.class ), mock( RequestContextFactory.class ),
                 Iterables.<KernelExtensionFactory<?>>empty(), mock( MasterClientResolver.class ),
                 ByteCounterMonitor.NULL, mock( RequestMonitor.class ), mock( SwitchToSlave.Monitor.class ),
-                StoreCopyMonitor.NONE ) );
+                new Monitors().newMonitor( StoreCopyClient.Monitor.class ) ) );
```
Comment (davidegrohmann) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/SwitchToSlaveTest.java]: perhaps you can simply create an empty monitor here: `new StoreCopyClient.Monitor.Adapter()`

**Comment by davidegrohmann** [comment]
@tinwelint it looks very nice

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/SwitchToSlaveTest.java (hunk 1 above)
Comment (tinwelint) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/SwitchToSlaveTest.java]: Sure, although it's a test so it doesn't really matter in this case.

Comment (davidegrohmann) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/SwitchToSlaveTest.java]: sure, it improves only readability :)

**Comment by zhenlineo** [comment]
Have you guys verified the problem only related to 2.2? Do we need to start from earlier versions?

**Comment by tinwelint** [comment]
I think this is only related to 2.2. The fact that this test doesn't actally test what it's set out to test may affect earlier versions. For earlier versions this would probably be a rewrite so that could come later.

**Comment by tinwelint** [comment]
Merge this in when opportunity arises

**Pull request merged**